ocaml/xapi/repository_helpers.ml

```diff
@@ -600,27 +648,49 @@ let get_installed_pkgs () =
   let params = ["-a"; "--pkgnarrow=installed"; "--qf"; fmt] in
   Helpers.call_script !Xapi_globs.repoquery_cmd params
   |> Astring.String.cuts ~sep:"\n"
+  |> List.map (fun x ->
+         debug "repoquery installed: %s" x ;
+         x
+     )
   |> List.fold_left parse_line_of_repoquery []
   |> List.map (fun (pkg, _) -> (Pkg.to_name_arch_string pkg, pkg))
 
-let get_updates_from_repoquery repositories =
+let run_repoquery pkg_narrow enabled_repos =
   let fmt = get_repoquery_fmt () in
   let params =
     [
       "-a"
     ; "--plugins"
     ; "--disablerepo=*"
-    ; Printf.sprintf "--enablerepo=%s" (String.concat "," repositories)
-    ; "--pkgnarrow=updates"
+    ; Printf.sprintf "--enablerepo=%s" enabled_repos
+    ; Printf.sprintf "--pkgnarrow=%s" pkg_narrow
     ; "--qf"
     ; fmt
     ]
   in
-  List.iter (fun r -> clean_yum_cache r) repositories ;
   Helpers.call_script !Xapi_globs.repoquery_cmd params
   |> Astring.String.cuts ~sep:"\n"
+  |> List.map (fun x ->
+         debug "repoquery available: %s" x ;
+         x
+     )
   |> List.fold_left parse_line_of_repoquery []
 
+let get_updates_from_repoquery repositories =
+  List.iter (fun r -> clean_yum_cache r) repositories ;
+  let enabled_repos = String.concat "," repositories in
```

Comment:
I think the naming is a bit redundant. Why do we need `enabled_repos` when there are not other repos in this code? Again, it might be better to use sets to express this logic.